R/utils.R

```diff
@@ -327,4 +335,66 @@ valid_seq <- function(params) {
   } else {
     return(FALSE)
   }
-}
\ No newline at end of file
+}
+
+resolve_prefix <- function(prefix, environment_var) {
+  if (!(is.null(prefix))) {
+    assertthat::assert_that(is.character(prefix))
+    
+    return(prefix)
+  } else {
+    prefix_env <- Sys.getenv(environment_var)
+    if (prefix_env != "") {
+      return(prefix_env)
+    } else {
+      return("/")
+    }
+  }
+}
+
+get_package_mapping <- function(script_path, url_package, dependencies) {
+  package_map <- vapply(dependencies, function(x) {
+    if (x$name %in% c('react', 'react-dom')) {
+      x$name <- 'dash_renderer'
+    }
+    dep_path <- paste(x$src$file, 
+                      x$script,
+                      sep = "/")
+  
+    # remove double slash and replace with single if present
+    result <- c(pkg_name=x$package,
+                dep_name=x$name,
+                dep_path=gsub("//", replacement = "/", dep_path)
+    )
+  }, FUN.VALUE = character(3))
+  
+  package_map <- as.data.frame(t(package_map), stringsAsFactors = FALSE)
+  
+  filename <- basename(script_path)
```

Comment:
does `basename` handle cross-platform directories? (i.e. `\` on windows)

Reply:
Yes, if I understand correctly. From the `basename` man page:

> *Behaviour on Windows*
>
>On Windows this will accept either \ or / as the path separator, but `dirname` will return a path using / (except if on a network share, when the leading \\ will be preserved). Expect these only to be able to handle complete paths, and not for example just a network share or a drive.
>
> UTF-8-encoded path names not valid in the current locale can be used.

This post explains why I'm not able to get `basename` to work as I would like when I pass it `C:\testpath\bundle.js`, since I'm using R in a *nix environment:

https://stat.ethz.ch/pipermail/r-help/2012-August/333592.html

>Yes, basename(filepath) assumes that filepath represents path to a file on the OS that R is running on.  Hence, on Windows we get
```  
> basename("dir\\path")
  [1] "path"
  > basename("dir/path")
  [1] "path"
```
>because both `/` and `\` are valid separators (and invalid in file names)
>
>Linux allows `\` in a file name so we get
```
> basename("dir\\file")
[1] "dir\\file"
> basename("dir/file")
[1] "file"
```